### DISCUSSION TITLE
Issue with brainvisa vipmatching command

### DISCUSSION THREAD

**Discussion by MartBony** [brainvisa/brainvisa.github.io#174] [Q&A]
Hi, I am trying to run the following command that executes brainvisa's vipmatching function inside a docker container : 

docker run --rm --user $ (id -u):$ (id -g) \
-v /tmp/tmp3k05o06q:/reference \
-v /home/my.folder/Apps/Slicer_modules/ModulesSTIM/StereoRefs/Resources/frameAIMs:/mobing \
-v /tmp/tmp3k05o06q:/inverse \
brainvisa-5.1.1:5.3 \
VipMatching \
    -r /reference/aux.nii.gz \
    -m /mobing/StereoFrameTemplate.Leksell.3voxels.S16.nii.gz \
    -R 't' \
    -inv /inverse/t1AimsRef2FrameAimsRef.trm

The command processes the inverse transformation but fails to write any files inside of the /inverse/ folder. Here is the full output : 

Reading reference surface...
Reading mobile surface...
Computing initial optimal translation...
Computing initial optimal translation...
Initialization using the surface centers of mass.
Computation of the distance map to the reference surface...
Forward.....
Slice:160
Backward.....
Slice:  1
Forward.....
Slice: 11
Backward.....
Slice:  1
Quadratric distance average: 55.4394
Surface matching...
With inflation: Warning: Simulated gradient
Optimal dilation in the gradient direction: -20.000 mm
With outlier elimination...
Rotation angle of the optimal transformation: 29.57 deg
\------------------------------
!! VIP Error:
       Can not open file for writing: StereoFrameTemplate.Leksell.3voxels.S16.nii.gz_TO_aux.nii.gz
\------------------------------
\------------------------------
!! VIP Error:
Execution aborted in function 'VipWriteDeplacement'.
\------------------------------
\------------------------------
!! VIP Error:
       Error while writting direct transformation file
\------------------------------
\-----------------------------
!! VIP Error:
Execution aborted in function '(commandline)VipWriteDeplacement'.
\------------------------------

I am not sure of what is the file it's struggling with, where it is trying to write it, or whether this is a permission issue.

**Comment by denisri** [comment]
Hi,
Just to reject any trivial explanation: is the /inverse directory writeable, and has it any free space/quota left ? Can you write a small file in it, like `echo something > /inverse/some_file`, from the docker container ?

**Comment by MartBony** [comment]
Hi @denisri,
I managed to make it work. I had to explicitly specify a path for the direct transformation (the file that was failing to be written) by adding the parameter : -dir /inverse/name.tfm
By default the full path for this is StereoFrameTemplate.Leksell.3voxels.S16.nii.gz_TO_aux.nii.gz so no directory is specified really, which may explain why it failed to write the file.

Thank you
